crates/ide/src/completion/presentation.rs

```diff
@@ -130,7 +137,18 @@ impl Completions {
             }
         }
 
-        completion_item.kind(kind).set_documentation(docs).add_to(self)
+        let final_completion_item = completion_item.kind(kind).set_documentation(docs);
+        if let Some((mutability, score)) = ref_matches {
+            let refed_completion_item = final_completion_item.clone();
+            refed_completion_item
+                .label(match mutability {
+                    RefMatch::Immutable => format!("&{}", local_name),
+                    RefMatch::Mutable => format!("&mut {}", local_name),
+                })
+                .set_score(score)
+                .add_to(self)
```

Comment:
Hm, I wonder if rather than cloning the item here, we perhaps should *flag* it as auto[mut]refable, and split this into to at the UI layer. 

What would the idal sorting scheme for `foo` and `&foo`  look like? Will we always try to place them side by side? 

API-wise, I worry about the introduction of Clone -- I think a better pattern would be to enhance a single item with meta, rather than duplicate items.

Reply:
Yeah that makes sense - but this mean each editor needs to implement the split itself, right?

Reply:
No, we need to implement the split at the LSP layer in `handlers.rs` or `to_proto.rs`.

Reply:
Got it!

Reply:
Moved the split to `completion_item` in `to_proto.rs`.